### DISCUSSION TITLE
what can be done to revive the eclipse-metals plugin?

### DISCUSSION THREAD

**Discussion by normana400** [scalameta/metals#6820] [General]
what needs to be done to revive the eclipse-metals project integration? Would love to have the scala metals in a full featured true IDE.

(don't know if this is the right place to ask this, but the eclipse plugin is archived in read-only and can't have the discussion there)

**Comment by normana400** [comment]
I use Theia and MSCode currently but I would really like to have the robustness of the old Scala IDE eclipse that worked in the Scala 12 / Java 8 days. The metals integration for eclipse never finished to a point where the integration is functional

**Comment by tgodzik** [comment]
Thanks for the question! We archived the plugin because it never reached a satisfactory level and we never had a long time maintainer. To revive that plugin we would need a maintainer that can take care of it, as we will not have time to work on it unfortunately

**Comment by normana400** [comment]
Is it better to restart the eclipse integration from scratch? Any idea how far along the old integration made it?

From: Tomasz Godzik ***@***.***>
Date: Friday, October 4, 2024 at 12:39 PM
To: scalameta/metals ***@***.***>
Cc: Andrew Norman ***@***.***>, Author ***@***.***>
Subject: [EXT] Re: [scalameta/metals] what can be done to revive the eclipse-metals plugin? (Discussion #6820)

  This message was sent from someone outside the company. If the sender is unknown, do not click on any links
  or open any attachments unless you recognize the source of this email and know the content is safe.



Thanks for the question! We archived the plugin because it never reached a satisfactory level and we never had a long time maintainer. To revive that plugin we would need a maintainer that can take care of it, as we will not have time to work on it unfortunately

**Reply by tgodzik** [reply]
> Is it better to restart the eclipse integration from scratch? Any idea how far along the old integration made it? 

I think the existing code can provide some hints on how to proceed, but it might need to be reworked in general
